src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -3,46 +3,99 @@
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3 for details about signed HTTP requests.
+        /// </remarks>
+        public PoPAuthenticationConfiguration()
         {
             ConfidentialClientApplication.GuardMobileFrameworks();
-            RequestUri = requestUri ?? throw new ArgumentNullException(nameof(requestUri));
         }
 
         /// <summary>
-        /// An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri (to the host and query parts of it)
+        /// Creates a configuration using the default key management, and which binds all the details of the HttpRequestMessage.
         /// </summary>
-        public Uri RequestUri { get; }
+        /// <remarks>
+        /// Currently only the HtppMethod (m), UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(HttpRequestMessage httpRequestMessage)
+        {
+            if (httpRequestMessage == null)
+                throw new ArgumentNullException(nameof(httpRequestMessage));
+
+            HttpMethod = httpRequestMessage.Method;
+            HttpHost = httpRequestMessage.RequestUri.Authority; // this includes the optional port
+            HttpPath = httpRequestMessage.RequestUri.AbsolutePath; 
+        }
+
+      
+
+        /// <summary>
+        /// Creates a configuration using the default key management, and which binds only the Uri part of the HTTP request.
+        /// </summary>
+        /// <remarks>
+        /// The UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(Uri requestUri)
+        {
+            if (requestUri == null)
+                throw new ArgumentNullException(nameof(requestUri));
+
+            HttpHost = requestUri.Authority;
+            HttpPath = requestUri.AbsolutePath;
+        }
 
         /// <summary>
         /// The HTTP method ("GET", "POST" etc.) method that will be bound to the token. Leave null and the POP token will not be bound to the method.
+        /// Corresponds to the "m" part of the a signed HTTP request. Optional.
         /// </summary>
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-3
+        /// </remarks>
         public HttpMethod HttpMethod { get; set; }
 
+        /// <summary>
+        /// The URL host of the protected API. The "u" part of a signed HTTP request.  This MAY include the port separated from the host by a colon in host:port format. Optional.
+        /// </summary>
+        public string HttpHost { get; set; }
```

Comment:
Public properties to allow full customization.

Reply:
Thought: Should we make set protected? To prevent any accidental update and security holes. With protected, developer will have to explicitly inherit from the the class and set it.

Reply:
This is more of a data-object type. We haven't used inheritance as extensibility mechanism in MSAL so far. 

From a security perspective, protected is exactly the same as public. See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/protected-members